optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/plugin/rest/LocationResource.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.vehiclerouting.plugin.rest;
+
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+import org.optaweb.vehiclerouting.domain.Coordinates;
+import org.optaweb.vehiclerouting.plugin.rest.model.PortableLocation;
+import org.optaweb.vehiclerouting.service.location.LocationService;
+
+@Path("location")
+public class LocationResource {
+
+    private final LocationService locationService;
+
+    @Inject
+    public LocationResource(LocationService locationService) {
+        this.locationService = locationService;
+    }
+
+    /**
+     * Create new location.
+     *
+     * @param request new location description
+     */
+    @Transactional
```

Comment:
TODO find out whether `@Transactional` belongs here. I suspect it might go in the service or even in the repository.